### PR TITLE
Updated tbDEX Wallet Quickstart

### PR DESCRIPTION
**Pull request by acekyd**
This pull request includes several significant updates to the `site/code-snippets/tbdex/wallet-quickstart.js` file and related documentation and tests. The changes focus on refactoring the wallet quickstart process, updating the credential schema, and improving the test suite.

### Refactoring and Updates to Wallet Quickstart:

* **Refactoring DID Creation and Import:**
  - Replaced hard-coded DID strings with dynamic DID creation using `DidDht.create` and removed the import of portable DIDs.

* **Credential Handling:**
  - Updated the verifiable credential (VC) application process to fetch a credential from an external service instead of creating it locally.
  - Adjusted the selection process for customer credentials to match the new schema requirements.

* **Simplification and Streamlining:**
  - Removed redundant functions and simplified the remaining ones, such as combining credential application and selection steps. [[1]](diffhunk://#diff-8ad10420a4e94ab2b88d8f03e6a284bec564a774d4e0ba1ec142b7802e80af20L9-R102) [[2]](diffhunk://#diff-8ad10420a4e94ab2b88d8f03e6a284bec564a774d4e0ba1ec142b7802e80af20R131-R135) [[3]](diffhunk://#diff-8ad10420a4e94ab2b88d8f03e6a284bec564a774d4e0ba1ec142b7802e80af20R144-R154) [[4]](diffhunk://#diff-8ad10420a4e94ab2b88d8f03e6a284bec564a774d4e0ba1ec142b7802e80af20R178-R179)

### Documentation Updates:

* **Credential Schema Update:**
  - Updated the documentation to reflect the new credential schema and issuer requirements. This includes changes to the `requiredClaims` section.

### Test Suite Enhancements:

* **Test Refactoring:**
  - Refactored the test suite to align with the updated wallet quickstart code, including dynamic DID creation and updated credential handling. [[1]](diffhunk://#diff-5c5f190947e356be03bd306d42ef83e30f2bf12b95965972fbcf62192bacba54L6-L98) [[2]](diffhunk://#diff-5c5f190947e356be03bd306d42ef83e30f2bf12b95965972fbcf62192bacba54R107-R150) [[3]](diffhunk://#diff-5c5f190947e356be03bd306d42ef83e30f2bf12b95965972fbcf62192bacba54L158-R160)
  - Added new test cases to ensure the correct functionality of the updated code, such as creating and submitting orders and processing quotes. [[1]](diffhunk://#diff-5c5f190947e356be03bd306d42ef83e30f2bf12b95965972fbcf62192bacba54L6-L98) [[2]](diffhunk://#diff-5c5f190947e356be03bd306d42ef83e30f2bf12b95965972fbcf62192bacba54R107-R150)

These changes collectively improve the maintainability, clarity, and functionality of the wallet quickstart process.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208137864726605

### CODE REVIEW

File: site/code-snippets/tbdex/wallet-quickstart.js [hunk 1]
```diff
@@ -6,160 +6,100 @@ import { VerifiableCredential } from '@web5/credentials'
...
+    context.customerDid = await DidDht.create({
+        options: {
+            publish: true,
```
Comment (angiejones) [inline site/code-snippets/tbdex/wallet-quickstart.js:11]: why publish?

File: site/code-snippets/tbdex/wallet-quickstart.js [hunk 2]
```diff
@@ -6,160 +6,100 @@ import { VerifiableCredential } from '@web5/credentials'
...
-    let customerDidString = '{"uri":"did:dht:h8e3yqnhgjwhtkjhxwhfy5mmkn4nebqxr8idguwrsxgef6ow8efo","document":{"id":"did:dht:h8e3yqnhgjwhtkjhxwhfy5mmkn4nebqxr8idguwrsxgef6ow8efo","verificationMethod":[{"id":"did:dht:h8e3yqnhgjwhtkjhxwhfy5mmkn4nebqxr8idguwrsxgef6ow8efo#0","type":"JsonWebKey","controller":"did:dht:h8e3yqnhgjwhtkjhxwhfy5mmkn4nebqxr8idguwrsxgef6ow8efo","publicKeyJwk":{"crv":"Ed25519","kty":"OKP","x":"4dGQOFwyacipPH04UG1rULQkBc8h6jNOhLPMgvoUOgs","kid":"XPokllC3LZAVGizIL0naDdByQHeyY12uLJaXO4j46Nw","alg":"EdDSA"}}],"authentication":["did:dht:h8e3yqnhgjwhtkjhxwhfy5mmkn4nebqxr8idguwrsxgef6ow8efo#0"],"assertionMethod":["did:dht:h8e3yqnhgjwhtkjhxwhfy5mmkn4nebqxr8idguwrsxgef6ow8efo#0"],"capabilityDelegation":["did:dht:h8e3yqnhgjwhtkjhxwhfy5mmkn4nebqxr8idguwrsxgef6ow8efo#0"],"capabilityInvocation":["did:dht:h8e3yqnhgjwhtkjhxwhfy5mmkn4nebqxr8idguwrsxgef6ow8efo#0"]},"metadata":{"published":true,"versionId":"1718740086"},"privateKeys":[{"crv":"Ed25519","d":"iTfn3Z8uPp3gTg-9LxQVZVODGqnP3M0UDjZiIwBEctc","kty":"OKP","x":"4dGQOFwyacipPH04UG1rULQkBc8h6jNOhLPMgvoUOgs","kid":"XPokllC3LZAVGizIL0naDdByQHeyY12uLJaXO4j46Nw","alg":"EdDSA"}]}'
+    context.customerDid = await DidDht.create({
```
Comment (angiejones) [inline site/code-snippets/tbdex/wallet-quickstart.js:9]: any reason to create a new one every time someone does the quickstart vs just using an existing one?

File: site/code-snippets/tbdex/wallet-quickstart.js [hunk 3]
```diff
@@ -6,160 +6,100 @@ import { VerifiableCredential } from '@web5/credentials'
...
+                  offering.data.payout.currencyCode === 'KES'
+          );
+      }
 
-    try {
-        const offerings  = await TbdexHttpClient.getOfferings({ pfiDid: context.pfiDid });
-    
-        context.selectedOffering = offerings[0];
-        console.log(context.selectedOffering);
+      console.log(context.selectedOffering);
```
Comment (angiejones) [inline site/code-snippets/tbdex/wallet-quickstart.js:31]: is the log needed?

File: site/code-snippets/tbdex/wallet-quickstart.js [hunk 4]
```diff
@@ -6,160 +6,100 @@ import { VerifiableCredential } from '@web5/credentials'
...
-        issuer  : issuer.uri,
-        subject : context.customerDid.uri,
-        data    : {
-            'beep': 'boop'
-        }
-        })
-        
-    const vcJwt = await vc.sign({ did: issuer})
+export async function quickstartApplyForCredential() {
+    const vcJwt = await fetch(`https://mock-idv.tbddev.org/kcc?name=angie&country=US&did=${context.customerDid.uri}`).then(res => res.text());
```
Comment (angiejones) [inline site/code-snippets/tbdex/wallet-quickstart.js:40]:
```suggestion
    const vcJwt = await fetch(`https://mock-idv.tbddev.org/kcc?name=alice&country=US&did=${context.customerDid.uri}`).then(res => res.text());
```

File: site/code-snippets/tbdex/wallet-quickstart.js [hunk 5]
```diff
@@ -6,160 +6,100 @@ import { VerifiableCredential } from '@web5/credentials'
...
     const selectedCredentials = PresentationExchange.selectCredentials({
-        vcJwts: myCredentials,
+        vcJwts: customerCredentials,
         presentationDefinition: context.selectedOffering.data.requiredClaims,
     });
-    // :snippet-end:
+
     context.credentials = selectedCredentials;
+
+    console.log("the selection", context.credentials);
```
Comment (angiejones) [inline site/code-snippets/tbdex/wallet-quickstart.js:57]: needed?

File: site/code-snippets/tbdex/wallet-quickstart.js [hunk 6]
```diff
@@ -188,25 +128,30 @@ async function processQuote() {
...
+    return context.order;
 }
 
-async function submitOrder() {
-    await context.order.sign(context.customerDid);
-    await TbdexHttpClient.submitOrder(context.order);
+export async function quickstartSendOrder() {
+  await context.order.sign(context.customerDid);
+  await TbdexHttpClient.submitOrder(context.order);
+  return "HTTP 202: Successfully Submitted Order";
```
Comment (angiejones) [inline site/code-snippets/tbdex/wallet-quickstart.js:151]: can we print out what's actually returned? do we get a status?

File: site/docs/tbdex/wallet-quickstart.mdx [hunk 7]
```diff
@@ -63,45 +62,61 @@ Once you have all of the PFI's offerings, you can filter it based on a variety o
...
             {
-              'path': [ 0: '$.issuer' ],
-              'filter': {
-                'type': 'string'
-                'const': 'did:dht:hge69zswp474myt94d149pftycsgk6yr9tufrh7new48re76b5ny'
+              "id": "73b86039-d07e-4f9a-9f3d-a8f7a8ec1635",
+              "constraints": {
+                "fields": [
+                  {
+                    "path": [
```
Comment (angiejones) [inline site/docs/tbdex/wallet-quickstart.mdx:84]: need both paths for JS?

Comment (acekyd) [reply site/docs/tbdex/wallet-quickstart.mdx:84]: No we don't I think that was PTSD 😓

File: site/code-snippets/tbdex/wallet-quickstart.js (hunk 2 above)
Comment (acekyd) [reply site/code-snippets/tbdex/wallet-quickstart.js:9]: I think I wanted to keep the full flow to simulate what the users would go through - from creating a DID and testing it out fresh. But I can revert if the former is preferred.

File: site/code-snippets/tbdex/wallet-quickstart.js (hunk 6 above)
Comment (acekyd) [reply site/code-snippets/tbdex/wallet-quickstart.js:148]: We don't get anything returned. Return type is void so we have to show our own message if all successful. The other one was in a try catch, will do the same for this.

Comment (angiejones) [reply site/code-snippets/tbdex/wallet-quickstart.js]: ok let's take out the http status part of the message if that's not what actually happens. can just say "Submitted Order"

File: site/code-snippets/tbdex/wallet-quickstart.js [hunk 8]
```diff
@@ -6,160 +6,100 @@ import { VerifiableCredential } from '@web5/credentials'
...
-async function sendRfq() {
+export async function quickstartSendRfq() {
     try{
         context.rfq.verifyOfferingRequirements(context.selectedOffering);
         await context.rfq.sign(context.customerDid);
         await TbdexHttpClient.createExchange(context.rfq);
     } catch (e) {
         throw e;
     }
+    return "HTTP 202: Successfully Submitted RFQ";
```
Comment (angiejones) [inline site/code-snippets/tbdex/wallet-quickstart.js]:
```suggestion
    return "Submitted RFQ";
```

File: site/code-snippets/tbdex/wallet-quickstart.js (hunk 6 above)
Comment (angiejones) [inline site/code-snippets/tbdex/wallet-quickstart.js]:
```suggestion
  return "Submitted Order";
```

File: site/code-snippets/tbdex/wallet-quickstart.js [hunk 9]
```diff
@@ -6,160 +6,98 @@ import { VerifiableCredential } from '@web5/credentials'
...
 
-async function sendRfq() {
-    try{
+export async function quickstartSendRfq() {
+    try {
         context.rfq.verifyOfferingRequirements(context.selectedOffering);
         await context.rfq.sign(context.customerDid);
         await TbdexHttpClient.createExchange(context.rfq);
+
+        return "HTTP 202: Successfully Submitted RFQ";
```
Comment (angiejones) [inline site/code-snippets/tbdex/wallet-quickstart.js:92]:
```suggestion
        return "Submitted RFQ";
```

File: site/code-snippets/tbdex/wallet-quickstart.js [hunk 10]
```diff
@@ -188,25 +126,35 @@ async function processQuote() {
...
+    return context.order;
 }
 
-async function submitOrder() {
+export async function quickstartSendOrder() {
+  try{
     await context.order.sign(context.customerDid);
     await TbdexHttpClient.submitOrder(context.order);
+
+    return "HTTP 202: Successfully Submitted Order";
```
Comment (angiejones) [inline site/code-snippets/tbdex/wallet-quickstart.js:151]:
```suggestion
    return "Submitted Order";
```

**Review by angiejones — APPROVED** [review]

**Pull request merged**